minidump-processor/src/op_analysis.rs

```diff
@@ -154,222 +234,341 @@ mod amd64 {
-                    Operand::DisplacementU32(disp) => Some(MemoryAccess {
-                        address: disp.into(),
-                        is_likely_null_pointer_dereference: false,
-                        is_likely_guard_page: false,
-                        size: mem_size,
-                    }),
-                    Operand::DisplacementU64(disp) => Some(MemoryAccess {
-                        address: disp,
-                        is_likely_null_pointer_dereference: false,
-                        is_likely_guard_page: false,
-                        size: mem_size,
-                    }),
-                    other_operand => {
-                        if let Some(op_info) = RegOperandInfo::try_from_operand(other_operand) {
-                            Some(self.calculate_address(mem_size, op_info)?)
-                        } else {
-                            None
-                        }
-                    }
-                };
+            Self::add_common_opcode_implicit_access(accesses, opcode, mem_size, context)?;
+            Ok(())
+        }
+
+        fn add_common_opcode_explicit_access(
+            accesses: &mut Vec<MemoryAccess>,
+            opcode: CommonOpcode,
+            operand: Operand,
+            idx: u8,
+            mem_size: Option<u8>,
+            context: &MinidumpContext,
+        ) -> Result<(), OpAnalysisError> {
+            if !operand.is_memory() {
+                return Ok(());
+            }
 
-                if let Some(access) = maybe_access {
-                    accesses.push(access);
+            let access_type = match (opcode, idx) {
+                (CommonOpcode::ADD | CommonOpcode::SUB, 0) => MemoryAccessType::ReadWrite,
+                (CommonOpcode::ADD | CommonOpcode::SUB, 1) => MemoryAccessType::ReadWrite,
+                (
+                    CommonOpcode::CALL
+                    | CommonOpcode::JMP
+                    | CommonOpcode::JMPF
+                    | CommonOpcode::PUSH,
+                    0,
+                ) => MemoryAccessType::Read,
+                (CommonOpcode::CMP, 0 | 1) => MemoryAccessType::Read,
+                (CommonOpcode::DEC | CommonOpcode::INC, 0) => MemoryAccessType::ReadWrite,
+                (CommonOpcode::MOV | CommonOpcode::MOVAPS | CommonOpcode::MOVUPS, 0) => {
+                    MemoryAccessType::Write
+                }
+                (CommonOpcode::MOV | CommonOpcode::MOVAPS | CommonOpcode::MOVUPS, 1) => {
+                    MemoryAccessType::Read
+                }
+                (CommonOpcode::LEA, 0 | 1) => return Ok(()),
+                _ => {
+                    tracing::warn!("instruction had unexpected memory operand");
+                    return Ok(());
                 }
+            };
+
+            if let Some(op_info) = MemoryOperandInfo::try_from_operand(operand) {
+                let address_info = MemoryAddressInfo::from_memory_operand_info(op_info, context)?;
```

Comment:
Ah ok this makes sense

Reply:
I will create `MemoryAddressInfo::try_from_operand()`, which calls `MemoryOperandInfo::try_from_operand()` and proceed with the logic of the original `MemoryAddressInfo::from_memory_operand_info()`. 
`MemoryOperandInfo::try_from_operand()` will remain a separate function since it is also used in `get_register()`